public/app/plugins/datasource/loki/LanguageProvider.ts

```diff
@@ -461,4 +516,38 @@ export default class LokiLanguageProvider extends LanguageProvider {
 
     return labelValues ?? [];
   }
+
+  async getLogInfo(selector: string): Promise<{ extractedLabelKeys: string[]; hasJSON: boolean; hasLogfmt: boolean }> {
+    const request: DataQueryRequest<LokiQuery> = {
+      requestId: 'LOKI_LOG_INFO',
+      interval: '1s',
+      intervalMs: 1000,
+      range: this.datasource.getTimeRange(),
+      scopedVars: {},
+      timezone: 'UTC', // FIXME
```

Comment:
@gabor Could you help me understand these `FIXME` here?

Reply:
@matyax at that point in the code, i need to create a `DataQueryRequest` to fetch some data from the database. so i have set up all the "unnecessary" attributes to some default-values to make typescript happy, but i added `FIXME` because i'm not sure if those values are really correct. for example, we should probably take the timezone-value from what is chosen in the grafana time-range-picker.

Reply:
but, that code might not be necessary anymore.. you see, the idea there was to fetch a couple lines of logs from Loki, "understand" those log-lines, and based on that offer smart autocomplete options about how to finish the logql-query.

but, some time ago the loki-query-builder got the query-hint functionality, that does basically the same.
for example, in the Loki query builder (assuming you are running `devenv` Loki), if you choose the labelName:`place`, labelValue:`luna` combination, it will offer you to add the `logfmt` parser. it does this by fetching a couple lines of logs and "understanding" them. this is the same as the one done by the monaco-loki-query-field, so i think we should be able to somehow "connect" to that functionality and use it too.

Reply:
Thank you for the explanation. I'll check if I can use that here.